src/Symfony/Component/DependencyInjection/ConfigurationResolverTrait.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection;
+
+use Symfony\Component\ExpressionLanguage\Expression;
+
+trait ConfigurationResolverTrait
+{
+    protected function resolve($value)
+    {
+        if (is_string($value) && '' !== $value && '@' === $value[0]) {
+            if (strlen($value) > 1 && $value[1] === '=') {
+                return new Expression(substr($value, 2));
```

Comment:
Since ES is a soft dep of DI, would it make sense to throw an exception here before triggering a fatal error when this syntax is used but ES is not available? We do so in other parts AFAIK